internal/generator/backend/template/representations/coeff.go.tmpl

```diff
@@ -27,6 +29,42 @@ func newCoeffTable(capacity int) CoeffTable {
 
 }
 
+func (ct *CoeffTable) toBytes() []byte {
+	buf := make([]byte, 0, 8 + len(ct.Coefficients)*fr.Bytes)
+	ctLen := uint64(len(ct.Coefficients))
+
+	buf = binary.LittleEndian.AppendUint64(buf, ctLen)
+	for _, c := range ct.Coefficients {
+		for _, w := range c {
+			buf = binary.LittleEndian.AppendUint64(buf, w)
+		}
+	}
+
+	return buf
+}
+
+func (ct *CoeffTable) fromBytes(buf []byte) error {
```

Comment:
yes so, bottom line is empirically, on large circuits, perf suffers by a significant factor (30-40% if I recall correctly without accounting for parallelism) using `io` `Reader` and `Writer`. It does a bit of reflection (or interface method access) behind the scene for example for `Uvarint` which is annoying ; plus doing []byte allows for parallel section processing. Wanted to maximize reading speed :) .

Second counter argument; yes it uses more memory than a streaming solution, but, 1GB will be for a 100M~ constraints circuits, I think if you are working with such large circuits, you can afford 1GB since you'll need 40times that anyway for the prover 😁